### PR TITLE
Document the Semantic Versioning guidelines

### LINKED ISSUE #458: Semver guidelines
Document Semver guidelines in the admin docs.

See https://central.owncloud.org/t/owncloud-core-switching-to-semver-apps-need-re-release/17054 for what version numbers will mean after OC 10.1

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #458.

### CODE REVIEW

File: modules/developer_manual/pages/core/semantic-versioning-guidelines.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+= Semantic Versioning Guidelines
+:semver_url: https://semver.org/
+
+The ownCloud core platform uses {semver_url}[Semantic Versioning (SemVer) 2.0.0].
```
Comment (PVince81) [inline modules/developer_manual/pages/core/semantic-versioning-guidelines.adoc]: should we add `since ownCloud 10.1.0` ?

Comment (settermjd) [reply modules/developer_manual/pages/core/semantic-versioning-guidelines.adoc]: Well said. I'll get that added.

**Comment by settermjd** [comment]
Change made, @PVince81. If you're happy with it, can you approve please, and I'll merge it.

**Comment by PVince81** [comment]
so later on we'll change the release URL to the final one ?

**Pull request merged**